fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/impl/object/builder/FabricTrackedDataRegistryImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.object.builder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.entity.data.TrackedDataHandler;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.Int2ObjectBiMap;
+
+import net.fabricmc.fabric.mixin.object.builder.TrackedDataHandlerRegistryAccessor;
+
+public class FabricTrackedDataRegistryImpl {
+	private static final Logger LOGGER = LoggerFactory.getLogger(FabricTrackedDataRegistryImpl.class);
+
+	private static List<TrackedDataHandler<?>> VANILLA_HANDLERS = new ArrayList<>();
+	private static SortedMap<Identifier, TrackedDataHandler<?>> MODDED_HANDLERS = new TreeMap<>();
+	private static List<TrackedDataHandler<?>> EXTERNAL_MODDED_HANDLERS = new ArrayList<>();
+
+	private FabricTrackedDataRegistryImpl() {
+	}
+
+	public static void storeVanillaHandlers() {
+		Int2ObjectBiMap<TrackedDataHandler<?>> dataHandlers = TrackedDataHandlerRegistryAccessor.fabric_getDataHandlers();
+
+		for (TrackedDataHandler<?> handler : dataHandlers) {
+			VANILLA_HANDLERS.add(handler);
+		}
+
+		LOGGER.debug("Stored {} vanilla handlers", VANILLA_HANDLERS.size());
+	}
+
+	/**
+	 * Reorders handlers in {@link net.minecraft.entity.data.TrackedDataHandlerRegistry#DATA_HANDLERS} to have a consistent order between client and server.
+	 *
+	 * <p>The order used is the following:
+	 *
+	 * <ul>
+	 *   <li>Vanilla handlers</li>
+	 *   <li>Handlers in the Fabric API registry (sorted by ID)</li>
```

Comment:
This is going to fall appart if you have diffrent handlers on the client and server. We have a similar issue with the blockstate ids, it works 99% of the time, but when it doesnt odd things happen.

I wonder if it would be better to actually sync the id -> int map during configuration?

Reply:
We discussed on Discord about possiblying registering a full Registry and then allowing reg sync to handle the syncing of the IDs. Im not sure how we would handle mods that arent using the API, but beyond that it shouldnt be too hard. Let me know if you want some help